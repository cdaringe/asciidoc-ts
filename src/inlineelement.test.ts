import { describe, expect, test } from "vitest";
import { toAST } from "./mod.js";

test("Sequential InlineElements", async ({ expect }) => {
  const input =
    `This paragraph has *bold*, _italic_, and ${"`"}monospace${"`"} text.`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});
