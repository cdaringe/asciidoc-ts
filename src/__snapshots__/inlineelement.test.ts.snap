// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Parse URL Macro 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "attributes": [
            "ExampleWebsite",
          ],
          "scheme": "https",
          "type": "UrlMacro",
          "url": "[semantics wrapper for AsciiDoc]://example.com",
        },
      ],
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;

exports[`Sequential InlineElements 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "This paragraph has ",
          "type": "PlainText",
        },
        {
          "content": [
            {
              "content": "bold",
              "type": "PlainText",
            },
          ],
          "type": "ConstrainedBold",
        },
        {
          "content": ", ",
          "type": "PlainText",
        },
        {
          "content": [
            {
              "content": "italic",
              "type": "PlainText",
            },
          ],
          "type": "UnconstrainedItalic",
        },
        {
          "content": ", and ",
          "type": "PlainText",
        },
        {
          "content": "monospace",
          "type": "MonospaceText",
        },
        {
          "content": "text.",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;
